### ISSUE TITLE
JBossClassLoaderHandler doesn't work with jboss-modules 1.5.x

### ISSUE THREAD

**Issue by jknack** [lukehutch/fast-classpath-scanner#90]
Hi,

The JBossClassLoaderHandler doesn't work with jboss-modules 1.5.x. Think the problem is an API change, for example I don't see `VFSResourceLoader` and `VirtualFile` classes.

They (guess) been replaced by `ResourceLoader` and `Resource`, also think something similar to this will make the `JBossClassLoaderHandler` works with 1.5.x:

```java
if ("org.jboss.modules.ModuleClassLoader".equals(loader.getClass().getName())) {
      // type ResourceLoader[]
      final Object vfsResourceLoaders = ReflectionUtils.invokeMethod(loader,
          "ResourceLoader");
      if (vfsResourceLoaders != null) {
        for (int i = 0, n = Array.getLength(vfsResourceLoaders); i < n; i++) {
          final Object resourceLoader = Array.get(vfsResourceLoaders, i);
          if (resourceLoader != null) {
            // type Resource
            final Object root = ReflectionUtils.invokeMethod(resourceLoader, "getResource",
                String.class, "/");
            if (root != null) {
              System.out.println(ReflectionUtils.invokeMethod(root, "getURL"));
            }
          }
        }
      }
    }
```

But I'm not 100% sure because there isn't a single way of testing my `custom class loader handler`, mainly because:

1. The JBossClassLoaderHandler is automatically registered.
2. The JBossClassLoaderHandler always return `true`. Even when no classpath entry was add it. This force process to stop and my custom classloader is never invoked.

Think this should be fixed in `JbossClassLoaderhandler` and ideally report `false` if no classpath entry was add it.

Thanks

**Comment by lukehutch** [comment]
Hey @jknack -- thanks for looking into this in such depth. I just committed a change that makes the default ClassLoaderHandlers return true only if at least one path element was added to the classpath. Hopefully this will allow you to register your own ClassLoaderHandler for this class. (Great suggestion.)

Yes, the way I'm pulling out classpath entries right now using reflection is an ugly hack, and this may not be the first time we'll need to adapt to API changes! I have very little knowledge of JBoss, so I really appreciate you looking into where to get the classpath entries in 1.5.x.

**Comment by jknack** [comment]
Great, thanks!

Is there a release date for 2.0.7?

**Comment by lukehutch** [comment]
There's no release date, but I can push out a release once this is fixed,
since this may affect a few users. I won't have time to look into this
myself for a week or two, but if you are able to submit a patch, I'll get
it out ASAP.

**Comment by lukehutch** [comment]
@jknack I notice you changed ```ReflectionUtils.invokeMethod(loader, "getResourceLoaders")``` into ```ReflectionUtils.invokeMethod(loader, "ResourceLoader")``` in your code, was this intentional?

Also, the reporter of bug #91 suggested looking at the following code for suggestions, is it worth calling out to the JBoss VFS framework in this way?:

https://github.com/StripesFramework/stripes/blob/master/stripes/src/main/java/net/sourceforge/stripes/vfs/JBoss6VFS.java

**Comment by jknack** [comment]
It was a typo (not intentional).

I'm not sure about `JBoss6VFS`. This is the **new** [ModuleClassLoader](https://github.com/jboss-modules/jboss-modules/blob/master/src/main/java/org/jboss/modules/ModuleClassLoader.java) that doesn't work with the existing `JBossClassLoaderHandler`. As you see there there is no reference to `vfs` or related classes.

Also, I do have a **custom** bootstrap. For example, I used it for [hot reload](http://jooby.org/doc/devtools/) of classes in my [micro-framework](http://jooby.org). To be honest, I'm not sure if my change is going to work in general and/or in `wildfly`.

For all these think the best is to release **as it is**, with the enhancement you already did.

Thoughts?

**Comment by lukehutch** [comment]
@jknack I think it's fine to add the change that gets this working for your use case. We'll add more changes as needed. Feel free to add your patch below the existing code, within the ```if (resourceLoader != null)``` block (i.e. leave the existing code there, and just add more URLs if they can be found using the introspection methods/fields you discovered). After you test it and show it's working, please send me a PR and I'll push out a release.

**Comment by jknack** [comment]
Done see #92

**Comment by lukehutch** [comment]
Fixed and pushed out as 2.0.7. Thank you, @jknack!
